project_test.go

```diff
@@ -61,6 +63,75 @@ func TestFindRoot(t *testing.T) {
 	}
 }
 
+func TestCheckCfgFilenames(t *testing.T) {
+	if runtime.GOOS != "windows" && runtime.GOOS != "darwin" {
+		t.Skip("skip this test on non-Windows, non-macOS")
+	}
+
+	errMsgFor := func(filetype, filename string) func(string) string {
+		return func(name string) string {
+			return fmt.Sprintf("%s filename %q does not match %q", filetype, name, filename)
+		}
+	}
+
+	manifestErrMsg := errMsgFor("manifest", ManifestName)
+	lockErrMsg := errMsgFor("lock", LockName)
+
+	invalidMfName := strings.ToLower(ManifestName)
+	invalidLfName := strings.ToLower(LockName)
+
+	cases := []struct {
+		wantErr     bool
+		createFiles []string
+		wantErrMsg  string
+	}{
+		// No error should be returned when the project contains a valid manifest file
+		// but no lock file.
+		{false, []string{ManifestName}, ""},
+		// No error should be returned when the project contains a valid manifest file as
+		// well as a valid lock file.
+		{false, []string{ManifestName, LockName}, ""},
+		// Error indicating the project was not found should be returned if a manifest
+		// file is not found.
+		{true, nil, errProjectNotFound.Error()},
+		// Error should be returned if the project has a manifest file with invalid name
+		// but no lock file.
+		{true, []string{invalidMfName}, manifestErrMsg(invalidMfName)},
+		// Error should be returned if the project has a valid manifest file and an
+		// invalid lock file.
+		{true, []string{ManifestName, invalidLfName}, lockErrMsg(invalidLfName)},
```

Comment:
What happens when the project has an `invalidMfName` and a valid `LockName`?

Reply:
The check is sequential. So if the manifest file name is invalid an error is returned and we do not check the lock file.